crates/core/src/keepers/common.rs

```diff
@@ -80,4 +86,46 @@ impl Keeper {
         dbg!(result);
         Ok(())
     }
+
+    /// Execute a withdrawal.
+    /// # Arguments
+    /// * `deposit_key` - The deposit key.
+    /// * `set_prices_params`.
+    pub async fn execute_withdrawal(
+        &self,
+        deposit_key: &str,
+        set_prices_params: &SetPricesParams,
+    ) -> Result<(), KeeperError> {
+        info!("running execute_deposit with key: {}", deposit_key);
+
+        let selector = get_selector_from_name("execute_withdrawal")
+            .map_err(|e| KeeperError::ConfigError(e.to_string()))?;
+
+        let deposit_key = FieldElement::from_hex_be(deposit_key).map_err(|e| {
+            KeeperError::ConfigError(format!(
+                "could not convert deposit_key '{}' into FieldElement: {}",
+                deposit_key, e
+            ))
+        })?;
+
+        // The deposit key
+        let mut calldata: Vec<FieldElement> = vec![deposit_key];
+        // The SetPricesParams arguments
+        calldata.extend::<Vec<FieldElement>>(set_prices_params.into());
+
+        let result = self
+            .account
+            .execute(vec![Call {
+                to: self.satoru_exchange_router_address,
+                selector,
+                calldata,
+            }])
+            .send()
+            .await
+            // TODO: Handle the error properly.
```

Comment:
i've opened an issue in order to add more details on error source.

Reply:
Ok nice! I'll let the TODO then & it would be removed when this issue is handled